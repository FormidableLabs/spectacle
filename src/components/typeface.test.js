import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';
import Typeface from './typeface';

class MockComponent extends Component {
  static propTypes = {
    children: PropTypes.node
  };

  static contextTypes = {
    typeface: PropTypes.object
  };

  render() {
    return <div style={this.context.typeface}>{this.props.children}</div>;
  }
}

describe('<Typeface />', () => {
  test('should render the children when using a system font.', () => {
    const wrapper = mount(
      <Typeface font="SF UI Text" weight={400}>
        <MockComponent>Hello!</MockComponent>
      </Typeface>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the children when using a Google font.', () => {
    const wrapper = mount(
      <Typeface googleFont="Roboto Slab" weight={700} italic>
        <MockComponent>Hello!</MockComponent>
      </Typeface>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
