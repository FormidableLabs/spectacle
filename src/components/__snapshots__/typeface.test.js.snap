// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typeface /> should render the children when using a Google font. 1`] = `
<Typeface
  googleFont="Roboto Slab"
  italic={true}
  weight={700}
>
  <div>
    <GoogleFont
      typography={
        Object {
          "options": Object {
            "googleFonts": Array [
              Object {
                "name": "Roboto Slab",
                "styles": Array [
                  "700i",
                ],
              },
            ],
          },
          "title": "Roboto Slab",
        }
      }
    >
      <link
        href="//fonts.googleapis.com/css?family=Roboto+Slab:700i"
        rel="stylesheet"
        type="text/css"
      />
    </GoogleFont>
    <MockComponent>
      <div
        style={
          Object {
            "fontFamily": "Roboto Slab",
            "fontStyle": "italic",
            "fontWeight": 700,
          }
        }
      >
        Hello!
      </div>
    </MockComponent>
  </div>
</Typeface>
`;

exports[`<Typeface /> should render the children when using a system font. 1`] = `
<Typeface
  font="SF UI Text"
  weight={400}
>
  <div>
    <MockComponent>
      <div
        style={
          Object {
            "fontFamily": "SF UI Text",
            "fontStyle": "normal",
            "fontWeight": 400,
          }
        }
      >
        Hello!
      </div>
    </MockComponent>
  </div>
</Typeface>
`;
